internal/k8s/client.go

```diff
@@ -0,0 +1,166 @@
+	}
+	for _, resource := range GetDeployments(clientset, namespace).Items {
+		resources = append(resources, resource.DeepCopyObject())
+	}
+	for _, resource := range GetPods(clientset, namespace).Items {
+		resources = append(resources, resource.DeepCopyObject())
+	}
+	for _, resource := range GetReplicationControllers(clientset, namespace).Items {
+		resources = append(resources, resource.DeepCopyObject())
+	}
+	for _, resource := range GetStatefulSets(clientset, namespace).Items {
+		resources = append(resources, resource.DeepCopyObject())
+	}
+	for _, resource := range GetNetworkPolicies(clientset, namespace).Items {
+		resources = append(resources, resource.DeepCopyObject())
+	}
+	for _, resource := range GetNamespaces(clientset, namespace).Items {
+		resources = append(resources, resource.DeepCopyObject())
+	}
+
+	return resources
+}
+
+func GetDaemonSets(clientset kubernetes.Interface, namespace string) *k8stypes.DaemonSetListV1 {
+	daemonSetClient := clientset.AppsV1().DaemonSets(namespace)
+	daemonSets, err := daemonSetClient.List(k8stypes.ListOptionsV1{})
+	if err != nil {
+		log.Error(err)
+	}
+	return daemonSets
+}
+
+func GetDeployments(clientset kubernetes.Interface, namespace string) *k8stypes.DeploymentListV1 {
+	deploymentClient := clientset.AppsV1().Deployments(namespace)
+	deployments, err := deploymentClient.List(k8stypes.ListOptionsV1{})
+	if err != nil {
+		log.Error(err)
+	}
+	return deployments
+}
+
+func GetPods(clientset kubernetes.Interface, namespace string) *k8stypes.PodListV1 {
+	podClient := clientset.CoreV1().Pods(namespace)
+	pods, err := podClient.List(k8stypes.ListOptionsV1{})
+	if err != nil {
+		log.Error(err)
+	}
+	return pods
+}
+
+func GetReplicationControllers(clientset kubernetes.Interface, namespace string) *k8stypes.ReplicationControllerListV1 {
+	replicationControllerClient := clientset.CoreV1().ReplicationControllers(namespace)
+	replicationControllers, err := replicationControllerClient.List(k8stypes.ListOptionsV1{})
+	if err != nil {
+		log.Error(err)
+	}
+	return replicationControllers
+}
+
+func GetStatefulSets(clientset kubernetes.Interface, namespace string) *k8stypes.StatefulSetListV1 {
+	statefulSetClient := clientset.AppsV1().StatefulSets(namespace)
+	statefulSets, err := statefulSetClient.List(k8stypes.ListOptionsV1{})
+	if err != nil {
+		log.Error(err)
+	}
+	return statefulSets
+}
+
+func GetNetworkPolicies(clientset kubernetes.Interface, namespace string) *k8stypes.NetworkPolicyListV1 {
+	netPolClient := clientset.NetworkingV1().NetworkPolicies(namespace)
+	netPols, err := netPolClient.List(k8stypes.ListOptionsV1{})
+	if err != nil {
+		log.Error(err)
+	}
+	return netPols
+}
+
+func GetNamespaces(clientset kubernetes.Interface, namespace string) *k8stypes.NamespaceListV1 {
+	namespaceClient := clientset.CoreV1().Namespaces()
+	listOptions := k8stypes.ListOptionsV1{}
+
+	if namespace != "" {
+		// Select only the specified namespace
+		// TODO this doesn't work??
```

Comment:
Concerning TODO here!

Reply:
Yes although I'm convinced it never worked so unless someone knows an easy fix or can confirm it actually does work I'll open a separate issue for it

Reply:
Could we open an issue to fix this? (see also https://github.com/Shopify/kubeaudit/pull/234#discussion_r366559035)

Reply:
https://github.com/Shopify/kubeaudit/issues/241